### PR TITLE
Bugfix FXIOS-7555 [v120] CFR shown on homepage

### PR DESCRIPTION
**Pull request by PARAIPAN9**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7555)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/16811)

## :bulb: Description

- Add condition to check if the app has started with the Homepage tab selected. If so, don't show the CFR to avoid showing it on the Homepage, intead of product detail.
## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [x] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by isabelrios** [comment]
The PR is failing because you can not run just a workflow. You need to Re-build either the entire pipeline or Re-build unsuccesful workflow. 
Triggering a single workflow for smoketest will fail because it does not have the dependencies it needs.

That said, you don't need to re-run the workflow. The failed UI Smoktest is due to a test that passed in a re-try. Only those that fail 3 out of 3 times will block the PR to merge.
In this case, there is only one failure and bitrise was green:
Screenshot 2023-10-16 at 15 28 49
https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/e6ebdf6e-dd22-42bf-89c5-03504cdb15e4

Unfortunately there is an issue and in that case the check in github is not updated. Once the PR is approved, it will be unblocked to merge

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift
Comment (PARAIPAN9) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift:133]: I've already tried this, but the problem is that for a short period of time, the currently selected tab is the product details URL. This is why the bug is happening. After that, a new blank homepage tab is created and selected as the currently used tab.

Comment (thatswinnie) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift:133]: That sounds like the root cause is that the selected tab is not correctly set in time and it should be fixed. While your solution is working I'd prefer having the root cause fixed.

Comment (PARAIPAN9) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift:133]: I can try that, but it's not necessarily a problem. When the app is opened, the last session of tabs is restored, making the latest used tab the current tab. If it happens to be the product detail page and the CFR needs to be shown, it shows the CFR. After that, it checks if 4 hours have passed, and if they have, another tab is created and selected as the current tab, leading to our bug. Maybe I could try and incorporate these two in one proccess.

Comment (OrlaM) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift:133]: This is a much bigger problem than just this ticket. We have an [epic](https://mozilla-hub.atlassian.net/browse/FXIOS-7314) to work on a more holistic solution for it. There are a bunch of issues around the app related to the timing of certain events on app launch or when returning from the background, most of them relating to the state of tabs. I'd recommend sticking with the hacky fix for now and we can swing back around once the epic above is completed. 

cc @mattreaganmozilla fyi

**Review by thatswinnie — APPROVED** [review]

**Pull request merged**
